site/src/xServices/sshKey/sshKeyXService.ts

```diff
@@ -0,0 +1,125 @@
+import { getUserSSHKey, regenerateUserSSHKey } from "api/api"
+import { GitSSHKey } from "api/typesGenerated"
+import { displaySuccess } from "components/GlobalSnackbar/utils"
+import { createMachine, assign } from "xstate"
+import { i18n } from "i18n"
+
+const { t } = i18n
+
+interface Context {
+  sshKey?: GitSSHKey
+  getSSHKeyError?: unknown
+  regenerateSSHKeyError?: unknown
+}
+
+type Events =
+  | { type: "REGENERATE_SSH_KEY" }
+  | { type: "CONFIRM_REGENERATE_SSH_KEY" }
+  | { type: "CANCEL_REGENERATE_SSH_KEY" }
+
+export const sshKeyMachine = createMachine(
+  {
+    id: "sshKeyState",
+    predictableActionArguments: true,
+    schema: {
+      context: {} as Context,
+      events: {} as Events,
+      services: {} as {
+        getSSHKey: {
+          data: GitSSHKey
+        }
+        regenerateSSHKey: {
+          data: GitSSHKey
+        }
+      },
+    },
+    tsTypes: {} as import("./sshKeyXService.typegen").Typegen0,
+    initial: "gettingSSHKey",
+    states: {
+      gettingSSHKey: {
+        entry: "clearGetSSHKeyError",
+        invoke: {
+          src: "getSSHKey",
+          onDone: [
+            {
+              actions: "assignSSHKey",
+              target: "loaded",
+            },
+          ],
+          onError: [
+            {
+              actions: "assignGetSSHKeyError",
+              target: "notLoaded",
+            },
+          ],
+        },
+      },
+      notLoaded: {
+        type: "final",
```

Comment:
Thinking about our conversation about simplifying XState machines by having less granularity: instead of making a separate error, `getSSHKeyError`, can we instead look at `sshState.matches("notLoaded")`?

Reply:
In this case we store this because we want to display the error.

Reply:
Oh, I didn't see the separate loading state - makes sense!